Remove localhost from code

lazzarello opened the pull request:
Small changes to the next.js setup configuration and documentation to allow the frontend and backend to be de-coupled without any change of functionality. This allows for the backend transcoder to be taken offline without the frontend going down with it.

Right now the frontend will display a message that "owncast is offline" which is true. This design could be improved.

Commit:
is this really all we need?

Commit:
is this really all we need?

Commit:
skip puppeteer

Commit:
new static files

Commit:
Merge branch 'remove-localhost-from-code' of github.com:lazzarello/chowncast into remove-localhost-from-code

Commit:
change modal language

Commit:
update doco

gabek commented:
It seems you checked in far more files than you intended to. Could you fix that up so it can be seen what was meant to change?

Commit:
add storybook example

lazzarello commented:
I ran a static build from `./build/web/bundleWeb.sh` not sure what to fix since this would change the static site, no?

gabek commented:
Don't check in any files under `static/web`. Those are generated.

lazzarello commented:
does that mean it should be under the .gitignore?

gabek commented:
CI will run a build and check in whatever is needed whenver changes to the web project get made. They shouldn't be ignored because when people clone the project they are required to run it.

gabek commented:
In general no human should ever need to run `./build/web/bundleWeb.sh`. If you do, and check in the changes, you get what you see above, a hundred conflicts.

lazzarello commented:
only 25 but I see your point. I suspected that was how a "new release" was built. I'll just replace them with the contents of HEAD at the develop branch.

Commit:
copy static from develop branch

lazzarello commented:
not sure where that `static/web/_next` modification came from. My local repo said I modified the following files after copying the whole static dir from develop.

```
        modified:   static/web/404.html
        modified:   static/web/404/index.html
        modified:   static/web/admin/access-tokens/index.html
        modified:   static/web/admin/actions/index.html
        modified:   static/web/admin/chat/emojis/index.html
        modified:   static/web/admin/chat/messages/index.html
        modified:   static/web/admin/chat/users/index.html
        modified:   static/web/admin/config-chat/index.html
        modified:   static/web/admin/config-federation/index.html
        modified:   static/web/admin/config-notify/index.html
        modified:   static/web/admin/config-social-items/index.html
        modified:   static/web/admin/config-video/index.html
        modified:   static/web/admin/config/general/index.html
        modified:   static/web/admin/config/server/index.html
        modified:   static/web/admin/federation/actions/index.html
        modified:   static/web/admin/federation/followers/index.html
        modified:   static/web/admin/hardware-info/index.html
        modified:   static/web/admin/help/index.html
        modified:   static/web/admin/index.html
        modified:   static/web/admin/logs/index.html
        modified:   static/web/admin/stream-health/index.html
        modified:   static/web/admin/upgrade/index.html
        modified:   static/web/admin/viewer-info/index.html
        modified:   static/web/admin/webhooks/index.html
        modified:   static/web/embed/chat/readonly/index.html
        modified:   static/web/embed/chat/readwrite/index.html
        modified:   static/web/embed/video/index.html
        modified:   static/web/index.html
        modified:   static/web/sw.js
```

The computer says there are no more merge conflicts!

gabek commented:
Did you mean to check in 10 files? That seems still more than what you intended to change.

Commit:
get next!

Commit:
Merge branch 'develop' into remove-localhost-from-code

Commit:
due dil·i·gence

lazzarello commented:
I think I fixed it. Can you check?

jprjr commented:
So to recap real quick: the core issue this is trying to solve - since owncast serves the pre-compiled front-end code, bringing the owncast process down results in the entire web app being unavailable.

I haven't tested this directly, but I think doing a split setup would break server-side hydration which is really useful for sharing links to the stream.

gabek commented on web/components/stores/ClientConfigStore.tsx:
```diff
@@ -38,7 +38,7 @@ let serverStatusRefreshPoll: ReturnType<typeof setInterval>;
 let hasWebsocketDisconnected = false;
 
-const serverConnectivityError = `Cannot connect to the Owncast service. Please check your internet connection and verify this Owncast server is running.`;
+const serverConnectivityError = `Cannot connect to the backend service. Please check your internet connection and verify this Owncast server is running.`;
```
This was specifically called out to be the "Owncast service" to be explicit about the error. To a end user, a viewer watching the stream, to tell them "the backend is down" doesn't mean anything to them.

gabek commented on test/automated/browser/run.sh:
```diff
@@ -32,6 +32,7 @@ fi
 # Install the web test framework
 if [ -z "$SKIP_BUILD" ]; then
 	echo "Installing test dependencies..."
+	export PUPPETEER_SKIP_DOWNLOAD=true
```
What is this?

gabek commented:
> So to recap real quick: the core issue this is trying to solve - since owncast serves the pre-compiled front-end code, bringing the owncast process down results in the entire web app being unavailable.
> 
> I haven't tested this directly, but I think doing a split setup would break server-side hydration which is really useful for sharing links to the stream.

The devil is in the details with all of this. I'll mention what I've been saying in the Owncast chat, I'm not interested in supporting a split frontend and backend setup, and there isn't any discussion about _how_ that were to work in the first place. I'm assuming it would mean that I'd have to ship separate frontend and backend binaries, and that's not something I'm willing to do.

Fixing the hardcoded `localhost`s to use the environment variable, however, is a good change.
